### PR TITLE
Cache qt

### PR DESCRIPTION
**Pull request by m-kuhn**
If all goes well that should speed up builds by a fair bit and remove a regularly failing dependency on external infrastructure (qt download servers)

### CODE REVIEW

File: .github/workflows/android.yml [hunk 1]
```diff
@@ -57,11 +56,12 @@ jobs:
 
       - name: 💐 Install Qt
         if: ${{ steps.vars.outputs.INSTALL_QT }}
-        uses: jurplel/install-qt-action@v2
+        uses: jurplel/install-qt-action@v3
         with:
           version: 5.14.2
           modules: 'qtcharts'
-          target: ${{ steps.vars.outputs.QT_TARGET }}
+          target: ' android'
```
Comment (nirvn) [inline .github/workflows/android.yml:63]: Is the extra space a typo?

```suggestion
          target: 'android'
```

File: .github/workflows/android.yml [hunk 2]
```diff
@@ -57,11 +56,12 @@ jobs:
 
       - name: 💐 Install Qt
         if: ${{ steps.vars.outputs.INSTALL_QT }}
-        uses: jurplel/install-qt-action@v2
+        uses: jurplel/install-qt-action@v3
         with:
           version: 5.14.2
           modules: 'qtcharts'
-          target: ${{ steps.vars.outputs.QT_TARGET }}
+          target: 'android'
```
Comment (nirvn) [inline .github/workflows/android.yml:63]:
```suggestion
          target: 'android'
          arch: 'android'
```

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 90ecb122c5f8ebe93f594537b57f48e79ae4b2b6: [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3190-c63d302c031491b4544f638af3053ba3d450e7ad-arm64-android.apk)
<details>
<summary>Other architectures</summary>

- [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3190-c63d302c031491b4544f638af3053ba3d450e7ad-arm-android.apk)
- [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3190-c63d302c031491b4544f638af3053ba3d450e7ad-x64-android.apk)
- [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3190-c63d302c031491b4544f638af3053ba3d450e7ad-x86-android.apk)
</details>

File: .github/workflows/android.yml (hunk 1 above)
Comment (m-kuhn) [reply .github/workflows/android.yml]: That was on purpose but didn't help.

File: .github/workflows/android.yml (hunk 2 above)
Comment (m-kuhn) [reply .github/workflows/android.yml:63]: good catch

Comment (nirvn) [reply .github/workflows/android.yml:63]: @m-kuhn , got the hint from this site (https://ddalcino.github.io/aqt-list-server/)

**Review by nirvn — APPROVED** [review]

**Comment by m-kuhn** [comment]
shaves 30-120(ios) seconds off build time

**Pull request merged**